packages/thunderstore-api/src/apiFetch.ts

```diff
@@ -39,32 +42,44 @@ async function fetchRetry(
   }
 }
 
+/**
+ * Simple timeout helper used by the retry loop.
+ */
 function sleep(delay: number) {
   return new Promise((resolve) => setTimeout(resolve, delay));
 }
 
-export type apiFetchArgs<B, QP> = {
+/**
+ * Arguments supplied to `apiFetch` describing the HTTP request and validation schemas.
+ */
+type SchemaOrUndefined<Schema extends z.ZodSchema | undefined> =
+  Schema extends z.ZodSchema ? z.infer<Schema> : undefined;
+
+export type apiFetchArgs<
+  RequestSchema extends z.ZodSchema | undefined,
+  QueryParamsSchema extends z.ZodSchema | undefined,
+> = {
   config: () => RequestConfig;
   path: string;
-  queryParams?: QP;
+  queryParams?: SchemaOrUndefined<QueryParamsSchema>;
   request?: Omit<RequestInit, "headers" | "body"> & { body?: string };
   useSession?: boolean;
-  bodyRaw?: B;
+  bodyRaw?: SchemaOrUndefined<RequestSchema>;
 };
 
-type schemaOrUndefined<A> = A extends z.ZodSchema
-  ? z.infer<A>
-  : never | undefined;
-
-export async function apiFetch(props: {
-  args: apiFetchArgs<
-    schemaOrUndefined<typeof props.requestSchema>,
-    schemaOrUndefined<typeof props.queryParamsSchema>
-  >;
-  requestSchema: z.ZodSchema | undefined;
-  queryParamsSchema: z.ZodSchema | undefined;
-  responseSchema: z.ZodSchema | undefined;
-}): Promise<schemaOrUndefined<typeof props.responseSchema>> {
+/**
+ * Validates input payloads, executes the HTTP request, and parses the response with Zod schemas.
+ */
+export async function apiFetch<
+  RequestSchema extends z.ZodSchema | undefined,
+  QueryParamsSchema extends z.ZodSchema | undefined,
+  ResponseSchema extends z.ZodSchema | undefined,
+>(props: {
+  args: apiFetchArgs<RequestSchema, QueryParamsSchema>;
+  requestSchema: RequestSchema;
+  queryParamsSchema: QueryParamsSchema;
+  responseSchema: ResponseSchema;
```

Comment:
Why are these three defined separately? If we have function `apiFetch` and a dedicated type `apiFetchArgs`, I'd expect the latter to define the actual arguments the former accepts, not just some subset of them. Nothing in the function body seems to benefit from having them defined separately either (e.g. passing `args` ahead to a helper function). Including these three in `args` would also reduce boilerplate code where undefined values are passed to the function.